jsHelper/spicetifyWrapper.js

```diff
@@ -1127,26 +1127,31 @@ Spicetify.SVGIcons = {
 })();
 
 class _HTMLContextMenuItem extends HTMLLIElement {
-	constructor({ name, disabled = false, icon = undefined, divider = false }) {
+	constructor({ name, disabled = false, icon = undefined, actionIcon = undefined, divider = false }) {
 		super();
 		this.name = name;
-		this.icon = icon || "";
+		(this.icon = icon), (this.actionIcon = actionIcon);
 		this.disabled = disabled;
 		this.divider = divider;
 		this.classList.add("main-contextMenu-menuItem");
 	}
+
 	render() {
-		let icon = this.icon;
-		if (icon && Spicetify.SVGIcons[icon]) {
-			icon = `<svg height="16" width="16" viewBox="0 0 16 16" fill="currentColor">${Spicetify.SVGIcons[icon]}</svg>`;
-		}
+		const icons = [this.icon, this.actionIcon];
+		icons.forEach((icon, index) => {
```

Comment:
you should use map instead but why you made it like this

Reply:
why not? it achieves the same thing without repeating code. I can use a map if you want but this works too

Reply:
just use map then

Reply:
```js
const icons = [this.icon, this.actionIcon];

const updatedIcons = icons.map((icon) => {
  if (icon && Spicetify.SVGIcons[icon]) {
    return `<svg height="16" width="16" viewBox="0 0 16 16" fill="currentColor">${Spicetify.SVGIcons[icon]}</svg>`;
  }
  return icon;
});
```

```js
const getIconHTML = (icon) => {
  if (icon && Spicetify.SVGIcons[icon]) {
    return `<svg height="16" width="16" viewBox="0 0 16 16" fill="currentColor">${Spicetify.SVGIcons[icon]}</svg>`;
  }
  return "";
};

const iconHTML = getIconHTML(this.icon);
const actionIconHTML = getIconHTML(this.actionIcon);
```

I'm a bit confused as to what you want here, these both are what i gather from what you are asking but seem way more complicated.

Reply:
> ```js
> const icons = [this.icon, this.actionIcon].map(icon =>
> 	icon && Spicetify.SVGIcons[icon] ? '<svg height="16" width="16" viewBox="0 0 16 16" fill="currentColor">${Spicetify.SVGIcons[icon]}</svg>' : icon || ""
> );
> ```

i believe the function method i did is a better solution, thoughts?

Reply:
that's why I resolved the conversation...

Reply:
oh sorry